### PR TITLE
Adds Abductor Janitorial Tools!

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Adds the following abductor janitorial items:
- Alien mop (never runs out of water, cleans faster than an advanced mop),
- Alien soap (syndicate soap, but abductor themed. Same speed, but no harsh chemicals!),
- Alien light replacer (can hold twice as many lights),
- Alien flyswatter (does an extra 1 damage),
- Alien janitorial belt (to hold the above items. Has one more slot than a normal janibelt.),
- Alien space cleaner (an abductor-themed advanced space cleaner bottle. Only accepts space cleaner.),
- Alien janitorial implant (janitor players rejoice!),
- Alien janitorial cyborg upgrade (janiborg players rejoice!).

The abductor UFO now has a filled alien janibelt in the upper locker. This does not affect the lavaland UFO crash site.

The janiborg now has a flyswatter by default (which is replaced with the alien version when given the alien upgrade).

NPF: Organises the abductor gear DMI into sections and groups everything together. Consolidates the engineering tools into the file.

## Why It's Good For The Game
Janitor enjoyers shall rejoice, for they shall receive more goodies.

## Images of changes
image

## Testing
Spawned in.
Verified sprites were all there. Verified the function of each and every item.
Verified that the belts correctly spawned in the abductor lockers.

## Changelog
:cl:
add: Adds abductor janitorial tools, also available in implant and cyborg upgrade form.
tweak: The janiborg module now has a flyswatter by default.
/:cl:

### CODE REVIEW

**Comment by DGamerL** [comment]
> Alien janitorial belt (to hold the above items. Has one more slot than a normal janibelt.),

Seems a bit meh for an abductor upgrade? Especially with there being an implant for it which does what the belt does but better

> The abductor scientist receives a filled toolbelt to keep the UFO free of grime.

Would fit better as an addition of their lockers honestly

**Comment by Fordoxia** [comment]
> > Alien janitorial belt (to hold the above items. Has one more slot than a normal janibelt.),
> 
> Seems a bit meh for an abductor upgrade? Especially with there being an implant for it which does what the belt does but better
> 
> > The abductor scientist receives a filled toolbelt to keep the UFO free of grime.
> 
> Would fit better as an addition of their lockers honestly

Belt has been moved to the top abductor ship lockers on the CC level.

**Comment by DGamerL** [comment]
> Belt has been moved to the top abductor ship lockers on the CC level.

That still doesn't mean the belt is something you want though? The implant does everything the belt does but better if I understand it?

**Comment by Fordoxia** [comment]
> > Belt has been moved to the top abductor ship lockers on the CC level.
> 
> That still doesn't mean the belt is something you want though? The implant does everything the belt does but better if I understand it?

You get access to the spraybottle and a holoprojector with the implant, but it requires RnD to manufacture. The abductors get the belt for free.

Abductor engineering tools are given to the agent in the form of a belt, but if the crew has the tech to make them, they'll pretty much always opt for the engi toolset implant which is also available at that tech level.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -1080,6 +1080,16 @@
 	construction_time = 12 SECONDS
 	category = list("Cyborg Upgrade Modules")
 
+/datum/design/borg_upgrade_abductor_jani
+	name = "Cyborg Upgrade (Abductor Janitorial Equipment)"
+	id = "borg_upgade_abductor_medi"
```
Comment (Henri215) [inline code/modules/research/designs/mechfabricator_designs.dm:1085]:
```suggestion
	id = "borg_upgade_abductor_jani"
```

File: code/modules/research/designs/medical_designs.dm [hunk 2]
```diff
@@ -425,6 +425,17 @@
 	build_path = /obj/item/organ/internal/cyberimp/arm/toolset_abductor
 	category = list("Medical")
 
+/datum/design/cyberimp_janitorial_abductor
+	name = "Abductor Toolset Implant"
+	desc = "An alien toolset, designed to be installed on subject's arm."
```
Comment (Henri215) [inline code/modules/research/designs/medical_designs.dm:430]: name and description are from the toolset implant, not janitorial one

File: code/modules/research/designs/medical_designs.dm [hunk 3]
```diff
@@ -425,6 +425,17 @@
...
+	id = "ci-jani-abductor"
+	req_tech = list("materials" = 6, "engineering" = 6, "biotech" = 6, "abductor" = 3)
+	build_type = PROTOLATHE | MECHFAB
+	materials = list(MAT_METAL = 20000, MAT_SILVER = 10000, MAT_PLASMA = 9000, MAT_TITANIUM = 8000, MAT_DIAMOND = 8000)
+	construction_time = 200
```
Comment (Henri215) [inline code/modules/research/designs/medical_designs.dm:435]:
```suggestion
	construction_time = 20 SECONDS
```

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 4]
```diff
@@ -652,21 +771,88 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	item_state = "flyswatter_abductor"
+	origin_tech = "abductor=1"
+	force = 2 // Twice as powerful thanks to alien technology!
+	throwforce = 2
+
+/obj/item/reagent_containers/spray/cleaner/safety/abductor	// Essentially an Advanced Space Cleaner, but abductor-themed. For the implant.
+	name = "alien space cleaner"
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "cleaner_abductor"
+	desc = "An alien spraybottle contaning alien-brand non-foaming space cleaner! It only accepts space cleaner."
```
Comment (Henri215) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:817]:
```suggestion
	name = "alien space cleaner"
	desc = "An alien spraybottle contaning alien-brand non-foaming space cleaner! It only accepts space cleaner."
	icon = 'icons/obj/abductor.dmi'
	icon_state = "cleaner_abductor"
```

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 5]
```diff
@@ -485,3 +485,24 @@
...
+	desc = "An experimental upgrade that replaces a janitorial cyborg's tools with the abductor versions."
+	icon_state = "abductor_mod"
+	origin_tech = "biotech=6;materials=6;abductor=3"
+	require_module = TRUE
+	module_type = /obj/item/robot_module/janitor
+	items_to_replace = list(
+		/obj/item/mop/advanced/cyborg = /obj/item/mop/advanced/abductor,
+		/obj/item/soap = /obj/item/soap/syndie/abductor,
+		/obj/item/lightreplacer/cyborg = /obj/item/lightreplacer/bluespace/abductor,
+		/obj/item/melee/flyswatter = /obj/item/melee/flyswatter/abductor,
```
Comment (Henri215) [inline code/game/objects/items/robot/robot_upgrades.dm:500]:
```suggestion
		/obj/item/melee/flyswatter = /obj/item/melee/flyswatter/abductor
```

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 6]
```diff
@@ -652,21 +771,88 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	name = "alien flyswatter"
+	desc = "For killing alien insects, obviously."
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "flyswatter_abductor"
+	item_state = "flyswatter_abductor"
+	origin_tech = "abductor=1"
+	force = 2 // Twice as powerful thanks to alien technology!
+	throwforce = 2
+
+/obj/item/reagent_containers/spray/cleaner/safety/abductor	// Essentially an Advanced Space Cleaner, but abductor-themed. For the implant.
```
Comment (Henri215) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:813]: image

For some reason its not using the correct mob sprite.

Comment (Fordoxia) [reply code/game/gamemodes/miniantags/abduction/abduction_gear.dm:813]: Fixed it.

**Review by Henri215 — COMMENTED** [review]
image

**Comment by Fordoxia** [comment]
> image

Fixed. The bucket wasn't checking if the item being added to it was a robot module.

File: code/game/objects/structures/mop_bucket.dm [hunk 7]
```diff
@@ -24,6 +24,9 @@
 	return ..()
 
 /obj/structure/mopbucket/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(W.is_robot_module())
+		to_chat(usr, "<span class='warning'>You cannot interface your modules with [src]!</span>")
```
Comment (Henri215) [inline code/game/objects/structures/mop_bucket.dm:28]:
```suggestion
		to_chat(user, "<span class='warning'>You cannot interface your modules with [src]!</span>")
```

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 8]
```diff
@@ -652,21 +771,89 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
-	blockTracking = 1
-	origin_tech = "materials=7;magnets=4;abductor=3"
-	flags = BLOCKHAIR
-	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES|HIDEFACE
+/////////////////////////////////////////
+//////////// JANITORIAL TOOLS ///////////
+/////////////////////////////////////////
+/obj/item/mop/advanced/abductor
+	name = "alien mop"
+	desc = "A collapsable mop clearly used by aliens to clean up any evidence of a close encounter. The head produces a constant supply of water when run over a surface, seemingly out of nowhere."
```
Comment (Henri215) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:779]:
```suggestion
	desc = "A collapsible mop clearly used by aliens to clean up any evidence of a close encounter. The head produces a constant supply of water when run over a surface, seemingly out of nowhere."
```

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 9]
```diff
@@ -652,21 +771,89 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	desc = "It's important to keep all the mysterious lights on a UFO functional when flying over backwater country."
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "lightreplacer_abductor"
+	origin_tech = "magnets=3;engineering=4;abductor=3"
+	max_uses = 40
+	uses = 20
+
+/obj/item/melee/flyswatter/abductor
+	name = "alien flyswatter"
+	desc = "For killing alien insects, obviously."
```
Comment (Henri215) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:805]: image

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 10]
```diff
@@ -652,21 +771,89 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "flyswatter_abductor"
+	item_state = "flyswatter_abductor"
+	origin_tech = "abductor=1"
+	force = 2 // Twice as powerful thanks to alien technology!
+	throwforce = 2
+
+/obj/item/reagent_containers/spray/cleaner/safety/abductor	// Essentially an Advanced Space Cleaner, but abductor-themed. For the implant.
+	name = "alien space cleaner"
+	desc = "An alien spraybottle contaning alien-brand non-foaming space cleaner! It only accepts space cleaner."
```
Comment (Henri215) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:815]:
```suggestion
	desc = "An alien spray bottle contaning alien-brand non-foaming space cleaner! It only accepts space cleaner."
```

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 11]
```diff
@@ -589,6 +433,281 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	requires_fuel = FALSE
+	refills_over_time = TRUE
+	low_fuel_changes_icon = FALSE
+
+/obj/item/crowbar/abductor
+	name = "alien crowbar"
+	desc = "A hard-light crowbar. It appears to pry by itself, without any effort required."
+	icon = 'icons/obj/abductor.dmi'
+	usesound = 'sound/weapons/sonic_jackhammer.ogg'
+	icon_state = "crowbar"
```
Comment (Henri215) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:661]:
```suggestion
	icon_state = "crowbar"
	usesound = 'sound/weapons/sonic_jackhammer.ogg'
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 12]
```diff
@@ -250,6 +250,17 @@
 /obj/item/organ/internal/cyberimp/arm/toolset_abductor/l
 	parent_organ = "l_arm"
 
+/obj/item/organ/internal/cyberimp/arm/janitorial_abductor
+	name = "Alien Janitorial Toolset implant"
```
Comment (Henri215) [inline code/modules/surgery/organs/augments_arms.dm:254]:
```suggestion
	name = "alien janitorial toolset implant"
```

**Review by Henri215 — APPROVED** [review]
LGTM

**Review by Burzah — APPROVED** [review]

**Pull request merged**